android/KMEA/app/src/main/java/com/tavultesoft/kmea/cloud/CloudDataJsonUtil.java

```diff
@@ -158,9 +158,10 @@ public static void processKeyboardPackageUpdateJSON(Context aContext, JSONObject
                 // Cloud catalog has newer KMP version available
                 String updateKMP = kbd.getUpdateKMP();
                 if (updateKMP != null) {
-                  if (!updateKMP.equalsIgnoreCase(cloudKMP)) {
-                    // Update keyboard info with the latest KMP link
-                    kbd.setUpdateKMP(cloudKMP);
+                  if (!updateKMP.startsWith(cloudKMP)) {
+                    // Update keyboard info with the latest KMP link after appending languageID
+                    String languageID = kbd.getLanguageID();
+                    kbd.setUpdateKMP(String.format("%s&bcp47=%s", cloudKMP, languageID));
```

Comment:
Technically, we should be encoding these urls. However, languageID is not likely to cause trouble as it uses only safechars, and cloudKMP also should be using only safechars. So I'm almost okay with it (except for defense-in-depth and pure principles!)